### DISCUSSION TITLE
Porting existing project to Nucleo-WL55

### DISCUSSION THREAD

**Discussion by super-cow-powers** [Lora-net/LoRaMac-node#1023] [Q&A] [answered]
Hi,
I wonder if anyone could give a few pointers on porting an existing LoRaMac-node project from the Murata Type ABZ to the new WL55/WLE5. 
I've had a look at the ST guide for porting to the WL, and it seems like a special case of STM32L4. Is there anything I should be particularly mindful of?

**Comment by nestorayuso** [comment]
it is almost done here:
https://www.st.com/content/st_com/en/products/embedded-software/mcu-mpu-embedded-software/stm32-embedded-software/stm32cube-mcu-mpu-packages/stm32cubewl.html

**Comment by elsalahy** [comment]
Yeah indeed, you don't need to port/implement anything, ST implementation already uses a slightly modified version of this stack.
See

https://github.com/STMicroelectronics/STM32CubeWL/tree/main/Projects/NUCLEO-WL55JC/Applications/LoRaWAN/LoRaWAN_End_Node

**Reply by super-cow-powers** [reply]
Thanks,
It is useful to compare notes, as it were - but I'm aiming to maintain compatibility in the application with mainline (and it's using functions and types from the Semtech HAL), so that we don't have to rely on ST to keep re-packaging it/to push updates in a timely manner, and so that we can build on multiple platforms if required.
Once we've a board from our rep, I'll try and get it set up like the other Nucleo Boards.

**Reply by elsalahy** [reply]
Yeah I understand, using LMN HAL can save time when it comes to porting your application to other MCUs/SOCs/SIPs.
Interestingly, the LMN HAL APIs are very similar to ST HAL APIs, so it should be fairly simple and straight forward, especially that you have a compatible stack.

**Reply by MarekNovakACRIOS** [reply]
Indeed - I have ported locally the ST support to this "mainline" version of the stack locally and it works ok - the only think I had to modify is the IRQ handling, which was IMHO quite race-condition prone....
I can prepare a pull request if someone is interested - I am currently deep into the platform in several of my projects.

Check #1036

**Reply by piyushy6** [reply]
> Yeah indeed, you don't need to port/implement anything, ST implementation already uses a slightly modified version of this stack.
> See
> 
> https://github.com/STMicroelectronics/STM32CubeWL/tree/main/Projects/NUCLEO-WL55JC/Applications/LoRaWAN/LoRaWAN_End_Node

Is there any way to increase the preamble length of transmission in the LoRaWAN_End_Node application?

**Comment by super-cow-powers** [comment, marked as answer]
Just going to mark this as answered. I glued together a port based on the ST version and the SX126x driver.
